resources/views/map/device-dependency.blade.php

```diff
@@ -97,7 +98,12 @@ function deviceSort(a,b) {
                 var keys = Object.keys(data).sort(deviceSort);
                 $.each( keys, function( dev_idx, device_id ) {
                     var device = data[device_id];
-                    var this_dev = {id: device_id, label: device["sname"], title: device["url"], shape: "box", level: device["level"]}
+
+                    // We need to pass a HTML element to title, otherwise it will intepret it as a string and not HTML
+                    let title = document.createElement("div");
+                    title.innerHTML = device["url"].replaceAll('img ', 'img loading="lazy" ');
```

Comment:
this is probably going to be a security vulnerability

Reply:
This was the bit I was going to suggest needs a review.

This is required to keep existing functionality, otherwise the pop-up stops working.  The addition of the <div> tag is the documented way that vis.js gives to keep the old functionality.

The device["url"] is the output of Url::deviceUrl(), which is why I think it's safe to add the loading="lazy" tag to all img tags.  A cleaner solution would be to adjust the Url::deviceUrl() code to include this tag (maybe even by default).

I think that the above code is safe given we generate the URL directly given only the device_id, but I am happy to change either the Url::deviceUrl() code, or even remove the pop-up graphs if you think either options is safer.

Reply:
Maybe generate the link via the component, which is at least a little safer. I think this is the code for that:

`Blade::render('<x-device-link :device="$device"></x-device-link>', ['device' => $device]);`

Reply:
Are you sure the above will work in a javascript loop?  From what I can see the Blade::render() needs the devices in a PHP loop.

Reply:
Or did you meant to call Blade::render in app/Http/Controllers/Maps/MapDataController.php?

I tried this, but the generated HTML didn't work.

Reply:
I think I figured it out and moved this to use blade code for generating the title